packages/data/src/test/registry.js

```diff
@@ -722,6 +722,40 @@ describe( 'createRegistry', () => {
 		} );
 	} );
 
+	describe( 'batch', () => {
+		it( 'should batch callbacks and only run the subscriber once', () => {
+			const store = registry.registerStore( 'myAwesomeReducer', {
+				reducer: ( state = 0 ) => state + 1,
+			} );
+			const listener = jest.fn();
+			subscribeWithUnsubscribe( listener );
+
+			registry.batch( () => {} );
+			expect( listener ).not.toHaveBeenCalled();
+
+			registry.batch( () => {
+				store.dispatch( { type: 'dummy' } );
+				store.dispatch( { type: 'dummy' } );
+			} );
+			expect( listener ).toHaveBeenCalledTimes( 1 );
+
+			const listener2 = jest.fn();
+			// useSelect subscribes to the stores differently,
+			// This test ensures batching works in this case as well.
```

Comment:
Probably worth to write some tests in `useSelect` then? To prevent testing the internals.

Reply:
I added a test to cover the behavior here which is not specific to useSelect but useSelect is the only user of `__experimentalSubscribeStore` at the moment.

Reply:
I would love to have some tests that actually render a component using `batch` and assert the render times using `Profiler` component. I think that would probably match how the API is used more?

Reply:
batch is lower level than this, it doesn't know anything about components or rerendering, so I think a lower level test like this one is actually better.

Reply:
But this doesn't actually test the description mentioned in the documentation though (render times for instance). I agree this test has its value, but I also think an additional higher-level test has benefits as well.

Reply:
I just tried to add a unit test using a component, the issue is that in order for the components to rerender in a unit test, you need to wrap them in a `act` call and the `act` has the same effect as `batch` meaning even if you remove `batch` call from the test, it still passes. So we can't actually test that behavior in a component unit test.